src/rgw/rgw_op.cc

```diff
@@ -2519,15 +2519,24 @@ void RGWPutObj::execute()
   }
 
   do {
-    bufferlist data;
-    len = get_data(data);
+    bufferlist data_in;
+    len = get_data(data_in);
     if (len < 0) {
       op_ret = len;
       goto done;
     }
     if (!len)
       break;
 
+    bufferlist data;
```

Comment:
With a reference we should get the expected behaviour without major changes. The next code works:

<pre>
-    bufferlist data;
+    bufferlist &data = data_in;
     if (s->aws4_auth_streaming_mode) {
       /* use unwrapped data */
       data = s->aws4_auth->bl;
       len = data.length();
-    } else {
-      data = data_in;
     }
</pre>

Does it make sense?